src/main/java/org/prebid/server/bidder/iqzone/IqZoneBidder.java

```diff
@@ -0,0 +1,93 @@
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class IqZoneBidder implements Bidder<BidRequest> {
+
+    private final JacksonMapper mapper;
+    private final String endpointUrl;
+
+    public IqZoneBidder(String endpointUrl, JacksonMapper mapper) {
+        this.mapper = Objects.requireNonNull(mapper);
+        this.endpointUrl = HttpUtil.validateUrl(endpointUrl);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        return Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .payload(request)
+                .body(mapper.encode(request))
+                .build());
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse), Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            throw new PreBidException("Bid response is null or seat bid is empty");
+        }
+        return bidsFromResponse(bidRequest, bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> constructBidderBid(bid, bidResponse))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private BidderBid constructBidderBid(Bid bid, BidResponse bidResponse) {
+        try {
+            return BidderBid.of(bid, getBidType(bid.getExt()), bidResponse.getCur());
+        } catch (IllegalArgumentException | PreBidException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private BidType getBidType(ObjectNode bidExt) {
```

Comment:
They use another approach for resolving bidType, most similar you can found in `ImprovedigitalBidder`